crates/iota-core/src/authority.rs

```diff
@@ -1882,9 +1887,7 @@ impl AuthorityState {
                 events,
                 object_changes,
                 balance_changes,
-                suggested_gas_price: self
-                    .congestion_tracker
-                    .get_suggested_gas_prices(&transaction),
+                suggested_gas_price,
```

Comment:
```suggestion
                suggested_gas_price: self
                    .congestion_tracker
                    .get_suggested_gas_price_with_ogd(&transaction),
```
Move it to the beginning before the `input` field, then you can avoid cloning transaction in the `input`.
If not clear, please check how this is done on the base branch.